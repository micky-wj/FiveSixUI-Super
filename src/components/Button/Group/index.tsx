import React from 'react';
import { getClassName, getKeyFromNode } from '../../_util/common';

export type ButtonSize = 'small' | 'large';

export interface ButtonGroupProps {
  className?: string;
  prefixCls?: string;
  size?: ButtonSize;
  style?: React.CSSProperties;
}

export default function ButtonGroup(props: ButtonGroupProps) {
  const { prefixCls = 'wl-btn-group', size = '', className, ...others } = props;

  // large => lg
  // small => sm
  const sizeCls = ({
    large: 'lg',
    small: 'sm',
  })[size] || '';

  // 注意这里className的设置 能够允许用户设置className
  const classes = `${className || ''} ` + getClassName(prefixCls, {
    [`${sizeCls}`]: !!sizeCls,
  });

  return <div className={classes} {...others}/>;
}
