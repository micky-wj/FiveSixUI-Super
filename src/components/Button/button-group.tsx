import React from 'react';
// import classNames from 'classnames';
import { getClassName, getKeyFromNode } from '../_util/common';

export type ButtonSize = 'small' | 'large';

export interface ButtonGroupProps {
  size?: ButtonSize;
  style?: React.CSSProperties;
  className?: string;
  prefixCls?: string;
}

export default function ButtonGroup(props: ButtonGroupProps) {
  const { prefixCls = 'ant-btn-group', size = '', className, ...others } = props;

  // large => lg
  // small => sm
  const sizeCls = ({
    large: 'lg',
    small: 'sm',
  })[size] || '';

  // const classes = getClassName(prefixCls, {
  //   [`${prefixCls}-${sizeCls}`]: sizeCls,
  // }, className);

  return <div {...others} className={className} />;
}
