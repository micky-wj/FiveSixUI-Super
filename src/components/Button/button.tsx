import React from 'react';
import { getClassName, getKeyFromNode } from '../_util/common';

import { findDOMNode } from 'react-dom';
// import Icon from '../icon';

const rxTwoCNChar = /^[\u4e00-\u9fa5]{2}$/;
const isTwoCNChar = rxTwoCNChar.test.bind(rxTwoCNChar);
function isString(str) {
  return typeof str === 'string';
}

// Insert one space between two chinese characters automatically.
function insertSpace(child) {
  // Check the child if is undefined or null.
  if (child == null) {
    return;
  }
  if (isString(child.type) && isTwoCNChar(child.props.children)) {
    return React.cloneElement(child, {},
      child.props.children.split('').join(' '));
  }
  if (isString(child)) {
    if (isTwoCNChar(child)) {
      child = child.split('').join(' ');
    }
    return <span>{child}</span>;
  }
  return child;
}

export type ButtonType = 'primary' | 'ghost' | 'dashed' | 'danger';
export type ButtonShape = 'circle' | 'circle-outline';
export type ButtonSize = 'small' | 'large';

export interface ButtonProps {
  type?: ButtonType;
  htmlType?: string;
  icon?: string;
  shape?: ButtonShape;
  size?: ButtonSize;
  onClick?: React.FormEventHandler<any>;
  onMouseUp?: React.FormEventHandler<any>;
  loading?: boolean;
  disabled?: boolean;
  style?: React.CSSProperties;
  prefixCls?: string;
  className?: string;
  ghost?: boolean;
}

export default class Button extends React.Component<ButtonProps, any> {
  static Group: any;

  static defaultProps = {
    prefixCls: 'wl-btn',
    type: 'default',
    size: 'default',
    loading: false,
    clicked: false,
    ghost: false,
  };

  static propTypes = {
    type: React.PropTypes.string,
    shape: React.PropTypes.oneOf(['circle', 'circle-outline']),
    size: React.PropTypes.oneOf(['large', 'default', 'small']),
    htmlType: React.PropTypes.oneOf(['submit', 'button', 'reset']),
    onClick: React.PropTypes.func,
    loading: React.PropTypes.bool,
    className: React.PropTypes.string,
    icon: React.PropTypes.string,
  };

  timeout: number;
  delayTimeout: number;

  constructor(props) {
    super(props);
    this.state = {
      loading: props.loading,
    };
  }

  componentWillReceiveProps(nextProps) {
    const currentLoading = this.props.loading;
    const loading = nextProps.loading;

    if (currentLoading) {
      clearTimeout(this.delayTimeout);
    }

    if (loading) {
      this.delayTimeout = setTimeout(() => this.setState({ loading }), 200);
    } else {
      this.setState({ loading });
    }
  }

  componentWillUnmount() {
    if (this.timeout) {
      clearTimeout(this.timeout);
    }
    if (this.delayTimeout) {
      clearTimeout(this.delayTimeout);
    }
  }

  handleClick = (e) => {
    // Add click effect
    const {onClick} = this.props;

    this.setState({ clicked: true });
    clearTimeout(this.timeout);
    this.timeout = setTimeout(() => this.setState({ clicked: false }), 500);

      onClick && onClick(e);
  }

  // Handle auto focus when click button in Chrome
  handleMouseUp = (e) => {
    (findDOMNode(this) as HTMLElement).blur();
    if (this.props.onMouseUp) {
      this.props.onMouseUp(e);
    }
  }

  render() {
    const { type, shape, size = '', htmlType, children, icon, className, prefixCls, ghost, ...others } = this.props;
    const { loading, clicked } = this.state;

    // large => lg
    // small => sm
    const sizeCls = ({
      large: 'lg',
      small: 'sm',
    })[size] || '';

    // 注意这里className的设置 能够允许用户设置className
    const classes = `${className} ` + getClassName(prefixCls, {
      [`${type}`]: !!type,
      [`${shape}`]: !!shape,
      [`${sizeCls}`]: !!sizeCls,
      'icon-only': !children && icon,
      loading,
      clicked,
      ghost,
    });

    const iconType = loading ? 'loading' : icon;
    const iconNode = iconType ? null : null;
    // <Icon type={iconType} />
    const kids = React.Children.map(children, insertSpace);

    return (
      <button
        type={htmlType || 'button'}
        className={classes}
        onMouseUp={this.handleMouseUp}
        onClick={this.handleClick}
      >
        {iconNode}{kids}
      </button>
    );
  }
}
