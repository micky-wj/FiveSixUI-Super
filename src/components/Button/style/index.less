@import '../../style/index.less';
@import "../../style/themes/default";
@import "../../style/mixins/index";
@import "./mixin";
@btn-prefix-cls: ~"@{wl-prefix}-btn";
// for compatibile
@btn-ghost-color : @text-color;
@btn-ghost-bg : transparent;
@btn-ghost-border : @border-color-base;
// Button styles
// -----------------------------
.@{btn-prefix-cls} {
    .btn;
    &-loading {
        pointer-events: none;
        .@{btn-prefix-cls}-loader {
            display: inline-block;
            position: relative;
            width: @font-size-base;
            height: @font-size-base;
            border: 1px solid fade(@primary-color, 80%);
            border-radius: 50%;
            animation: loading 1s ease-in-out infinite;
            -webkit-animation: loading 1s ease-in-out infinite;
            &:after {
                content: '';
                width: 4px;
                height: 4px;
                border-radius: 50%;
                background: @primary-color;
                position: absolute;
                top: -2px;
                left: 2px;
            }
        }
    }
    &-default {
        .btn-default;
    }
    &-primary {
        .btn-primary;
        &.@{btn-prefix-cls}-loading {
            background-color: lighten(@primary-color, 20%);
            border-color: lighten(@primary-color, 20%);
            .@{btn-prefix-cls}-loader {
                border-color: fade(#fff, 80%);
                &:after {
                    background: #fff;
                }
            }
        }
        .@{btn-prefix-cls}-group &:not(:first-child):not(:last-child) {
            border-right-color: @btn-group-border;
            border-left-color: @btn-group-border;
            &:disabled {
                border-color: @btn-default-border;
            }
        }
        .@{btn-prefix-cls}-group &:first-child {
            &:not(:last-child) {
                border-right-color: @btn-group-border;
                &[disabled] {
                    border-right-color: @btn-default-border;
                }
            }
        }
        .@{btn-prefix-cls}-group &:last-child:not(:first-child),
        .@{btn-prefix-cls}-group &+& {
            border-left-color: @btn-group-border;
            &[disabled] {
                border-left-color: @btn-default-border;
            }
        }
    }
    &-ghost {
        .btn-ghost;
    }
    &-dashed {
        .btn-dashed;
    }
    &-danger {
        .btn-danger;
        &.@{btn-prefix-cls}-loading {
            background-color: lighten(@btn-danger-bg, 20%);
            border-color: lighten(@btn-danger-bg, 20%);
            .@{btn-prefix-cls}-loader {
                border-color: fade(#fff, 80%);
                &:after {
                    background: #fff;
                }
            }
        }
    }
    &-circle,
    &-circle-outline {
        .btn-circle(@btn-prefix-cls);
        .@{btn-prefix-cls}-loader {
            margin-left: (@btn-circle-size - @font-size-base)/2-1;
            margin-right: (@btn-circle-size - @font-size-base)/2;
        }
        &.@{btn-prefix-cls}-lg .@{btn-prefix-cls}-loader {
            margin-left: (@btn-circle-size-lg - @font-size-base)/2-1;
            margin-right: (@btn-circle-size-lg - @font-size-base)/2;
        }
        &.@{btn-prefix-cls}-sm .@{btn-prefix-cls}-loader {
            margin-left: (@btn-circle-size-sm - @font-size-base)/2-1;
            margin-right: (@btn-circle-size-sm - @font-size-base)/2;
        }
    }
    &:not(&-circle):not(&-circle-outline)&-icon-only {
        padding-left: 8px;
        padding-right: 8px;
    }
    &:not(&-circle):not(&-circle-outline):not(&-icon-only) {
        .@{btn-prefix-cls}-loader {
            margin-right: 6px;
        }
    }
    &-group {
        .btn-group(@btn-prefix-cls);
        .@{btn-prefix-cls}-clicked {
            animation: none;
            transition: none;
        }
    }
    &-clicked {
        transition-timing-function: cubic-bezier(.6, 4, .3, .8);
        animation: bangbang .5s 1;
    }
    &-bg-ghost {
        background: transparent!important;
        border-color: #fff;
        color: #fff;
        &.disabled,
        &[disabled] {
            color: #c0c0c0;
            border-color: #c0c0c0;
            background: #343434!important;
        }
    }
    &-bg-ghost&-primary {
        .button-variant-transpbg(@primary-color);
    }
    &-bg-ghost&-danger {
        .button-variant-transpbg(@btn-danger-bg);
    }
    &-full-width {
        width: 100%;
    }
}

@keyframes bangbang {
    from,
    to {
        -webkit-transform: scale(1, 1);
        transform: scale(1, 1)
    }
    25% {
        -webkit-transform: scale(.9, 1.1);
        transform: scale(.9, 1.1)
    }
    50% {
        -webkit-transform: scale(1.1, .9);
        transform: scale(1.1, .9)
    }
    75% {
        -webkit-transform: scale(.95, 1.05);
        transform: scale(.95, 1.05)
    }
}

@-webkit-keyframes bangbang {
    from,
    to {
        -webkit-transform: scale(1, 1);
        transform: scale(1, 1)
    }
    25% {
        -webkit-transform: scale(.9, 1.1);
        transform: scale(.9, 1.1)
    }
    50% {
        -webkit-transform: scale(1.1, .9);
        transform: scale(1.1, .9)
    }
    75% {
        -webkit-transform: scale(.95, 1.05);
        transform: scale(.95, 1.05)
    }
}

@keyframes loading {
    0% {
        transform: rotate(0);
    }
    100% {
        transform: rotate(360deg);
    }
}

@-webkit-keyframes loading {
    0% {
        transform: rotate(0);
    }
    100% {
        transform: rotate(360deg);
    }
}