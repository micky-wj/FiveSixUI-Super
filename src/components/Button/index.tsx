import React from 'react';
import Group from './Group';
import { getClassName, getKeyFromNode } from '../_util/common';

import { findDOMNode } from 'react-dom';
// import Icon from '../icon';

const rxTwoCNChar = /^[\u4e00-\u9fa5]{2}$/;
const isTwoCNChar = rxTwoCNChar.test.bind(rxTwoCNChar);
function isString(str) {
    return typeof str === 'string';
}

// 中文字符间加入一个空格
function insertSpace(child) {
    if (child == null) {
        return;
    }
    if (isString(child.type) && isTwoCNChar(child.props.children)) {
        return React.cloneElement(child, {},
            child.props.children.split('').join(' '));
    }
    if (isString(child)) {
        if (isTwoCNChar(child)) {
            child = child.split('').join(' ');
        }
        return <span>{child}</span>;
    }
    return child;
}
export type ButtonHtmlType = 'submit' | 'button' | 'reset';
export type ButtonShape = 'circle' | 'circle-outline';
export type ButtonSize = 'small' | 'large';
export type ButtonType = 'default' | 'primary' | 'dashed' | 'danger' | 'ghost';

export interface ButtonProps {
    className?: string;
    icon?: string; // 暂不支持
    prefixCls?: string;
    disabled?: boolean;
    fullWidth?: boolean;
    ghost?: boolean;
    loading?: boolean;
    htmlType?: ButtonHtmlType;
    shape?: ButtonShape;
    size?: ButtonSize;
    type?: ButtonType;
    style?: React.CSSProperties;
    onClick?: React.FormEventHandler<any>;
    onMouseUp?: React.FormEventHandler<any>;
}

export default class Button extends React.Component<ButtonProps, any> {
    static Group = Group;
    static __FIVESIX_BUTTON = true;

    static defaultProps = {
        prefixCls: 'wl-btn',
        disabled: false,
        fullWidth: false,
        ghost: false,
        loading: false,
        htmlType: 'button',
        type: 'default',
    };

    static propTypes = {
        className: React.PropTypes.string,
        icon: React.PropTypes.string,
        // disabled: React.PropTypes.bool,
        // fullWidth: React.PropTypes.bool,
        // ghost: React.PropTypes.bool,
        // loading: React.PropTypes.bool,
        htmlType: React.PropTypes.oneOf(['submit', 'button', 'reset']),
        shape: React.PropTypes.string,
        size: React.PropTypes.string,
        type: React.PropTypes.string,
        style: React.PropTypes.object,
        onClick: React.PropTypes.func,
    };

    clickedTimer: number;
    loadedTimer: number;

    constructor(props) {
        super(props);
        this.state = {
            loaded: props.loading,
            clicked: false
        };
    }

    componentWillReceiveProps(nextProps) {
        const currentLoading = this.props.loading;
        const loaded = nextProps.loading;

        if (currentLoading) {
            clearTimeout(this.loadedTimer);
        }

        if (loaded) {
            this.loadedTimer = setTimeout(() => this.setState({ loaded }), 200);
        } else {
            this.setState({ loaded });
        }
    }

    componentWillUnmount() {
        if (this.clickedTimer) {
            clearTimeout(this.clickedTimer);
        }
        if (this.loadedTimer) {
            clearTimeout(this.loadedTimer);
        }
    }

    handleClick = (e) => {
        // 增加点击特效
        const { onClick } = this.props;

        this.setState({ clicked: true });
        clearTimeout(this.clickedTimer);
        this.clickedTimer = setTimeout(() => {
            this.setState({ clicked: false });
        onClick && onClick(e);

        }, 500);

    }

    // Handle auto focus when click button in Chrome
    handleMouseUp = (e) => {
        (findDOMNode(this) as HTMLElement).blur();
        if (this.props.onMouseUp) {
            this.props.onMouseUp(e);
        }
    }

    render() {
        const { className, prefixCls, shape, size = '', type, disabled,ghost, fullWidth, loading, icon, htmlType, children, ...others } = this.props;
        const { loaded, clicked } = this.state;

        const sizeCls = ({
            large: 'lg',
            small: 'sm',
        })[size] || '';

        // 注意这里className的设置 能够允许用户设置className
        const classes = `${className || ''} ` + getClassName(prefixCls, {
            [`${shape}`]: !!shape,
            [`${sizeCls}`]: !!sizeCls,
            [`${type}`]: !!type,
            'bg-ghost': !!ghost,
            'full-width': !!fullWidth,
            'loading': !!loaded,
            'icon-only': !children && (icon || loaded),
            clicked,
        });

        const loaderNode = loaded ? (<div className={`${prefixCls}-loader`}></div>) : '';
        const iconNode = icon ? null : '';
        // <Icon type={iconType} />
        const kids = React.Children.map(children, insertSpace);

        return <button
            type={htmlType}
            className={classes}
            disabled={!!disabled}
            onMouseUp={this.handleMouseUp}
            onClick={this.handleClick}
            {...others}
        >
            {loaderNode || iconNode}{kids}
        </button>;
    }
}
