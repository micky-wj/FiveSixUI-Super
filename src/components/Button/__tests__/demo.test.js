import React from 'react';
import Button from '..';
import renderer from 'react-test-renderer';

// 简陋版GUI测试
// 在最开始需要撰写简单的应用场景，在开发、迭代时会提示对界面的更改，以确保更改是符合预期的

const ButtonGroup = Button.Group;

test('Button demo basic', () => {
    const component = renderer.create(
        <div>
            <Button type="primary">Primary</Button>
            <Button>Default</Button>
            <Button type="dashed">Dashed</Button>
            <Button type="danger">Danger</Button>
            <Button type="ghost">Ghost</Button>
        </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Button demo disabled', () => {
    const component = renderer.create(
        <div>
            <Button type="primary">Primary</Button>
            <Button type="primary" disabled>Primary(disabled)</Button>
            <br />
            <Button>Default</Button>
            <Button disabled>Default(disabled)</Button>
            <br />
            <Button>Ghost</Button>
            <Button disabled>Ghost(disabled)</Button>
            <br />
            <Button type="dashed">Dashed</Button>
            <Button type="dashed" disabled>Dashed(disabled)</Button>
        </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Button demo ghost', () => {
    const component = renderer.create(
        <div style={{ background: 'rgb(190, 200, 200)', padding: '26px 16px 16px' }}>
            <Button type="primary" ghost>Primary</Button>
            <Button ghost>Default</Button>
            <Button type="dashed" ghost>Dashed</Button>
            <Button type="danger" ghost>danger</Button>
        </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Button demo loading', () => {
    const component = renderer.create(
        <div>
            <Button type="primary" loading>
                Loading
        </Button>
            <Button type="primary" size="small" loading>
                Loading
        </Button>
            <br />
            <Button shape="circle" loading />
            <Button type="primary" shape="circle" loading />
        </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Button demo size', () => {
    const component = renderer.create(
        <div>
            <Button type="primary" size="large">Large</Button>
            <Button type="primary" size="small">Small</Button>
        </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Button demo fullWidth', () => {
    const component = renderer.create(
        <div>
            <Button type="primary" fullWidth>fullWidth</Button>
            <Button type="primary">Normal</Button>
        </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Button demo ButtonGroup', () => {
    const component = renderer.create(
        <div>
            <h4>Basic</h4>
            <ButtonGroup>
                <Button>Cancel</Button>
                <Button type="primary">OK</Button>
            </ButtonGroup>
            <ButtonGroup>
                <Button disabled>L</Button>
                <Button disabled>M</Button>
                <Button disabled>R</Button>
            </ButtonGroup>
            <ButtonGroup>
                <Button type="primary">L</Button>
                <Button>M</Button>
                <Button>M</Button>
                <Button type="dashed">R</Button>
            </ButtonGroup>
        </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
