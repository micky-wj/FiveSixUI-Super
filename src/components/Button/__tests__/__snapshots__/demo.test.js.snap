exports[`test Button demo ButtonGroup 1`] = `
<div>
  <h4>
    Basic
  </h4>
  <div
    className=" wl-btn-group">
    <button
      className=" wl-btn wl-btn-default"
      disabled={false}
      onClick={[Function]}
      onMouseUp={[Function]}
      type="button">
      
      <span>
        Cancel
      </span>
    </button>
    <button
      className=" wl-btn wl-btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseUp={[Function]}
      type="button">
      
      <span>
        OK
      </span>
    </button>
  </div>
  <div
    className=" wl-btn-group">
    <button
      className=" wl-btn wl-btn-default"
      disabled={true}
      onClick={[Function]}
      onMouseUp={[Function]}
      type="button">
      
      <span>
        L
      </span>
    </button>
    <button
      className=" wl-btn wl-btn-default"
      disabled={true}
      onClick={[Function]}
      onMouseUp={[Function]}
      type="button">
      
      <span>
        M
      </span>
    </button>
    <button
      className=" wl-btn wl-btn-default"
      disabled={true}
      onClick={[Function]}
      onMouseUp={[Function]}
      type="button">
      
      <span>
        R
      </span>
    </button>
  </div>
  <div
    className=" wl-btn-group">
    <button
      className=" wl-btn wl-btn-primary"
      disabled={false}
      onClick={[Function]}
      onMouseUp={[Function]}
      type="button">
      
      <span>
        L
      </span>
    </button>
    <button
      className=" wl-btn wl-btn-default"
      disabled={false}
      onClick={[Function]}
      onMouseUp={[Function]}
      type="button">
      
      <span>
        M
      </span>
    </button>
    <button
      className=" wl-btn wl-btn-default"
      disabled={false}
      onClick={[Function]}
      onMouseUp={[Function]}
      type="button">
      
      <span>
        M
      </span>
    </button>
    <button
      className=" wl-btn wl-btn-dashed"
      disabled={false}
      onClick={[Function]}
      onMouseUp={[Function]}
      type="button">
      
      <span>
        R
      </span>
    </button>
  </div>
</div>
`;

exports[`test Button demo basic 1`] = `
<div>
  <button
    className=" wl-btn wl-btn-primary"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Primary
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-default"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Default
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-dashed"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Dashed
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-danger"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Danger
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-ghost"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Ghost
    </span>
  </button>
</div>
`;

exports[`test Button demo disabled 1`] = `
<div>
  <button
    className=" wl-btn wl-btn-primary"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Primary
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-primary"
    disabled={true}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Primary(disabled)
    </span>
  </button>
  <br />
  <button
    className=" wl-btn wl-btn-default"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Default
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-default"
    disabled={true}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Default(disabled)
    </span>
  </button>
  <br />
  <button
    className=" wl-btn wl-btn-default"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Ghost
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-default"
    disabled={true}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Ghost(disabled)
    </span>
  </button>
  <br />
  <button
    className=" wl-btn wl-btn-dashed"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Dashed
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-dashed"
    disabled={true}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Dashed(disabled)
    </span>
  </button>
</div>
`;

exports[`test Button demo fullWidth 1`] = `
<div>
  <button
    className=" wl-btn wl-btn-primary wl-btn-full-width"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      fullWidth
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-primary"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Normal
    </span>
  </button>
</div>
`;

exports[`test Button demo ghost 1`] = `
<div
  style={
    Object {
      "background": "rgb(190, 200, 200)",
      "padding": "26px 16px 16px",
    }
  }>
  <button
    className=" wl-btn wl-btn-primary wl-btn-bg-ghost"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Primary
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-default wl-btn-bg-ghost"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Default
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-dashed wl-btn-bg-ghost"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Dashed
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-danger wl-btn-bg-ghost"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      danger
    </span>
  </button>
</div>
`;

exports[`test Button demo loading 1`] = `
<div>
  <button
    className=" wl-btn wl-btn-primary wl-btn-loading"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    <div
      className="wl-btn-loader" />
    <span>
      Loading
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-sm wl-btn-primary wl-btn-loading"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    <div
      className="wl-btn-loader" />
    <span>
      Loading
    </span>
  </button>
  <br />
  <button
    className=" wl-btn wl-btn-circle wl-btn-default wl-btn-loading wl-btn-icon-only"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    <div
      className="wl-btn-loader" />
  </button>
  <button
    className=" wl-btn wl-btn-circle wl-btn-primary wl-btn-loading wl-btn-icon-only"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    <div
      className="wl-btn-loader" />
  </button>
</div>
`;

exports[`test Button demo size 1`] = `
<div>
  <button
    className=" wl-btn wl-btn-lg wl-btn-primary"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Large
    </span>
  </button>
  <button
    className=" wl-btn wl-btn-sm wl-btn-primary"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    
    <span>
      Small
    </span>
  </button>
</div>
`;
